src/http.rs

```diff
@@ -123,6 +124,17 @@ where
                 })
                 .context(CallRequestSnafu)?;
 
+            let (parts, body) = response.into_parts();
+            let mut buf: Vec<u8> = Vec::new();
+            let body = body.map_data(move |data| {
+                buf.extend(data);
+                if buf.len() > 20 {
+                    println!("Data buf has reached len 20: {:?}", std::str::from_utf8(&buf));
+                }
+                bytes::Bytes::new()
```

Comment:
I was facing an issue with the compiler complaining that `Data` was moved, in order to fix this, instead of cloning Data I only return an empty byte.

Is this fine for now?

Reply:
I should be able to use extend with a reference to data, vec has an extend from slice function and data can be referenced as a slice. Making us able to return Data instead of dropping it.